index.html

```diff
@@ -160,15 +238,36 @@
 
     points.enter().append('circle')
       .attr('class', 'point')
-      .attr('r', 5);
+      .attr('r', 2);
 
     points
-      .attr('cx', function (d) { return x(d.unemployment); })
-      .attr('cy', function (d) { return y(d.inflation); });
+      .attr('cx', function (a) { return x(a.costs); })
+      .attr('cy', function (a) { return y(a.visits); });
 
     points.exit().remove();
 
     // HISTOGRAMS
+    var map = txData.map(function(d) { return d.costs });
+
+    var histogram = d3.layout.histogram()
+                      .bins(x.ticks())
+                      (map);
+
+    var bars = svg.selectAll(".bar")
+                  .data(histogram)
+                  .enter().append("g")
+                  .attr("class", "bar"); 
+
+    histScale = d3.scale.linear()
+                        .domain([0, d3.max(bars, function (d) { return d.length; })])
+                        .range([0, height])
+                        .nice(); 
+
+    bars.append("rect")
+        .attr("x", 0)
+        .attr("y", 0)
```

Comment:
The `x` attribute is what's really tripping you up here. If it's always 0, where will all the bars be positioned?

Of course, the `y` attribute will also present some difficulties, but you'll realize that quickly once the `x` is fixed.